src/sentry/search/events/builder/discover.py

```diff
@@ -344,54 +332,16 @@ def resolve_query(
                 self.groupby = self.resolve_groupby(groupby_columns)
 
-    def load_config(
-        self,
-    ) -> tuple[
-        Mapping[str, Callable[[str], SelectType]],
-        Mapping[str, fields.SnQLFunction],
-        Mapping[str, Callable[[event_search.SearchFilter], WhereType | None]],
-        Mapping[str, Callable[[Direction], OrderBy]],
-    ]:
-        self.config: DatasetConfig
-        if self.dataset in [
-            Dataset.Discover,
-            Dataset.Transactions,
-            Dataset.Events,
-            Dataset.IssuePlatform,
-        ]:
-            self.config = DiscoverDatasetConfig(self)
-        elif self.dataset == Dataset.Sessions:
-            self.config = SessionsDatasetConfig(self)
-        elif self.dataset in [Dataset.Metrics, Dataset.PerformanceMetrics]:
-            if self.spans_metrics_builder:
-                # For now, we won't support the metrics layer for spans since it needs some work,
-                # but once the work will be done, we will have to add:
-                # if self.builder_config.use_metrics_layer:
-                #     self.config = SpansMetricsLayerDatasetConfig(self)
-                self.config = SpansMetricsDatasetConfig(self)
-            elif self.profile_functions_metrics_builder:
-                self.config = ProfileFunctionsMetricsDatasetConfig(self)
-            elif self.builder_config.use_metrics_layer:
-                self.config = MetricsLayerDatasetConfig(self)
-            else:
-                self.config = MetricsDatasetConfig(self)
-        elif self.dataset == Dataset.Profiles:
-            self.config = ProfilesDatasetConfig(self)
-        elif self.dataset == Dataset.Functions:
-            self.config = ProfileFunctionsDatasetConfig(self)
-        elif self.dataset == Dataset.SpansIndexed:
-            self.config = SpansIndexedDatasetConfig(self)
-        elif self.dataset == Dataset.MetricsSummaries:
-            self.config = MetricsSummariesDatasetConfig(self)
-        else:
-            raise NotImplementedError(f"Data Set configuration not found for {self.dataset}.")
-
-        field_alias_converter = self.config.field_alias_converter
-        function_converter = self.config.function_converter
-        search_filter_converter = self.config.search_filter_converter
-        orderby_converter = self.config.orderby_converter
+    def parse_config(self) -> None:
+        self.field_alias_converter = self.config.field_alias_converter
+        self.function_converter = self.config.function_converter
+        self.search_filter_converter = self.config.search_filter_converter
+        self.orderby_converter = self.config.orderby_converter
+        self.config_parsed = True
 
-        return field_alias_converter, function_converter, search_filter_converter, orderby_converter
+    def load_config(self) -> None:
+        self.config = self.config_class(self)
+        self.parse_config()
```

Comment:
This seems like an odd pattern to have the child class call `self.parse_config()` every time it implements `load_config`.

Can we try something like

```python
def load_config(self) -> DatasetConfig:
  return self.config_class(self)
  
def __init__(self, ...):
  # other stuff
  self.config = self.load_config()
  self.parse_config()
```

That way child classes can just return an instance of `DatasetConfig`?

Reply:
Agreed